lib/pages/sport_details_page.dart

```diff
@@ -132,6 +134,14 @@ class _SportDetailsPageState extends State<SportDetailsPage> {
                     itemCount: events.length,
                     itemBuilder: (context, index) {
                       final event = events[index];
+                      // Get current user ID
+                      final currentUserId =
+                          widget.authenticationBloc.sportUser?.uuid ?? '';
+
+                      // Check if the user is registered for this event
+                      final isRegistered =
+                          event.registeredUsers?.contains(currentUserId) ?? false;
```

Comment:
This should have been done by checking the registration collection like we agreed on Tuesday last week. For a many-to-many relationship, you should use a join table like the registration collection. 

I guess for now it's fine, but this is a bad design.

Reply:
it is checking the events collection table where registered users are stored